zebrad/src/application.rs

```diff
@@ -243,17 +243,19 @@ impl Application for ZebradApp {
 
         // Load config *after* framework components so that we can
         // report an error to the terminal if it occurs (unless used with a command that doesn't need the config).
-        let config = match command.config_path() {
-            Some(path) => match self.load_config(&path) {
-                Ok(config) => config,
-                // Ignore errors loading the config for some commands.
-                Err(_e) if command.cmd().should_ignore_load_config_error() => Default::default(),
-                Err(e) => {
-                    status_err!("Zebra could not parse the provided config file. This might mean you are using a deprecated format of the file. You can generate a valid config by running \"zebrad generate\", and diff it against yours to examine any format inconsistencies.");
-                    return Err(e);
-                }
-            },
-            None => ZebradConfig::default(),
+        let config = match ZebradConfig::load(command.config_path()) {
+            Ok(config) => config,
+            // Ignore errors loading the config for some commands.
+            Err(_e) if command.cmd().should_ignore_load_config_error() => Default::default(),
+            Err(e) => {
+                status_err!("Zebra could not load its configuration: {}", e);
```

Comment:
It would be nice to leave the extra guidance here.

```suggestion
                status_err!(
                    "Zebra could not load the provided configuration file and/or environment variables.\
                     This might mean you are using a deprecated format of the file, or are attempting to
                     configure deprecated or unknown fields via environment variables.\
                     You can generate a valid config by running \"zebrad generate\", \
                     and diff it against yours to examine any format inconsistencies."
                );
```